sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
```

Comment:
This is an interesting proposition. It would also entail much more work to implement some of these other methods ('auto', 'fd', 'doane', and 'scott'). Unfortunately we can't use this feature of `np.histogram` in scikit-learn. This feature was added to numpy in version `1.11.0`, while the minimum requirements of scikit-learn are `1.6.1`.